autonomy/deploy/image.py

```diff
@@ -57,115 +55,62 @@ class ImageProfiles:  # pylint: disable=too-few-public-methods
-                        "-p",
-                        profile,
-                    ],
-                    preexec_fn=os.setsid,
-                    env=env,
-                    stdout=subprocess.PIPE,
-                    cwd=str(skaffold_dir),
-                )
-            )
-
-            for line in iter(cast(IO[bytes], process.stdout).readline, ""):
-                if line == b"":
-                    break
-                print(f"[Skaffold] {line.decode().strip()}")
-
-        except KeyboardInterrupt:  # pragma: nocover
-            cast(IO[bytes], process.stdout).close()
-            process.send_signal(signal.SIGTERM)
-
-        process.wait(timeout=30)
-        poll = process.poll()
-        if poll is None:  # pragma: nocover
-            process.terminate()
-            process.wait(2)
-
-        if process.returncode != 0:  # pragma: nocover
-            print("Image build failed.")
-
-    @staticmethod
-    def _copy_packages(
-        packages_dir: Path,
-        build_dir: Path,
-    ) -> None:
-        """Copy packages for image building."""
-        build_packages_dir = Path(build_dir) / "packages"
-
-        if build_packages_dir.exists():
-            shutil.rmtree(build_packages_dir)
-
-        shutil.copytree(  # type: ignore # pylint: disable=E1123
-            src=Path(packages_dir), dst=build_packages_dir
+def build_image(  # pylint: disable=too-many-arguments
+    agent: PublicId,
+    profile: str,
+    skaffold_dir: Path,
+    version: str,
+    push: bool = False,
+    build_concurrency: int = 0,
+) -> None:
+    """Command to build images from for skaffold deployment."""
+
+    env = os.environ.copy()
+    env["AEA_AGENT"] = str(agent)
+
+    if profile == ImageProfiles.DEPENDENCIES:
+        env["VERSION"] = version
+    else:
+        env["VERSION"] = f"{agent.name}-{version}"
+
+    kubeconfig = env.get("KUBECONFIG")
+    if profile == ImageProfiles.CLUSTER:
+        if kubeconfig is None and not check_kubeconfig_vars():
+            raise ValueError("Please setup kubernetes environment variables.")
+    else:
+        if kubeconfig is not None:  # pragma: nocover
+            del env["KUBECONFIG"]
+
+    try:
+        process = subprocess.Popen(  # nosec # pylint: disable=consider-using-with,W1509
+            [
+                "skaffold",
+                "build",
+                f"--build-concurrency={build_concurrency}",
+                f"--push={str(push).lower()}",
+                "-p",
+                profile,
+            ],
+            preexec_fn=os.setsid,
+            env=env,
+            stdout=subprocess.PIPE,
+            cwd=str(skaffold_dir),
         )
 
-    @staticmethod
-    def get_aea_agent(deployment_file_path: Path) -> str:
-        """Validate and retrieve aea agent from spec."""
+        for line in iter(cast(IO[bytes], process.stdout).readline, ""):
+            if line == b"":
+                break
+            print(f"[Skaffold] {line.decode().strip()}")
```

Comment:
man can't wait till we can use the [walrus operator](https://realpython.com/python-walrus-operator/). But we need to stop supporting python3.7 first

Reply:
I think the walrus operator is only supported on >= 3.10, So we'll have to remove support for 3.8, 3.9 as well 😆